lib/archethic_web/controllers/api/web_hosting_controller.ex

```diff
@@ -84,6 +89,7 @@ defmodule ArchethicWeb.API.WebHostingController do
     with {:ok, address} <- Base.decode16(address, case: :mixed),
          true <- Crypto.valid_address?(address),
          {:ok, reference_transaction} <- ReferenceTransaction.fetch_last(address),
+         :ok <- published?(reference_transaction),
```

Comment:
if you name your function with `?` it should return a boolean to keep the Elixir's convention.
So you should rename it, probably.
You can maybe instead of returning a tuple, return the status right away, so maybe:  `:published <- publication_status(reference_transaction)`